scripts/lib-node.sh

```diff
@@ -1,6 +1,42 @@
 genesis_root="$(realpath configuration/genesis)"
 genesis_file="${genesis_root}/genesis.json"
 
+usage() {
+        cat <<EOF
+Usage:
+
+  $(basename $0) [ --cabal | --stack | --stack-nix ] *
+
+EOF
+}
+
+default_mode='cabal'
+mode='default'
+while test -n "$1"
+do case "$1" in
+           --cabal )              mode='cabal';;
+           --stack )              mode='stack';;
+           --stack-nix )          mode='stack-nix';;
+           --help | "--"* ) usage; exit 1;;
+           * ) break;; esac; shift; done
+if   test "$mode" != 'default'
+then true
+elif test -d 'dist-newstyle'
```

Comment:
I am not sure this will work. What if $CWD is not where 'dist-newstyle' is?
same for '.stack-work'.

Reply:
Done.